### ISSUE TITLE
how to Hazelcast jet JDBC continues source get data when new record insert or existing record updated in a database table?

### ISSUE THREAD

**Issue by nuriozalp** [hazelcast/hazelcast-jet-code-samples#170]
please urgent help,about how to Hazelcast jet JDBC continues source get data when new record insert or existing record updated in a database table?

**Comment by nuriozalp** [comment]
you are all the time say realtime elt and you can create connector but this
is the one of the most important data source and jdbc has already listeners
and you can easly implement listeners and you can activate windows
sliding.  i really want to use jet in big project but because of this
reason it will create some diffecult for most of elt operation. i will get
data from service bus now. if you have plan to add listeners also to
database tables, please let me know about your plan?

**Comment by viliam-durina** [comment]
JDBC doesn't provide a standardized way for listeners. There are many possible workarounds, such as using triggers, polling, db-specific ways. Therefore it's not easy to provide one-size-fits-all out-of-the-box solution. You can use `SourceBuilder` or, if you need feature that's not covered by it, implement a `Processor` directly.

Also, Jet absolutely is very close to real-time inside. But it also supports batch processing, which is the case for the bundled JDBC connector.

**Comment by nuriozalp** [comment]
yes but there are a few most common database and each of them can be seperated by db name. and in framework can decide whiich one should it to use.  i wanna know how can i write any task in pipeline and acording to my windows sliding i will ger my source sql query (new record and only updated record between specific time.) i want to use same structure of jet pipeline and source and join and sink structure but somehow i need to trigger this job with new input arguments? can show help for it ? maybe some code sample?

**Comment by cangencer** [comment]
@nuriozalp you could use a custom source / SourceBuilder to achieve this. A code sample which polls a HTTP endpoint is available here: https://github.com/hazelcast/hazelcast-jet-code-samples/tree/0.7-maintenance/source-builder
